Integrate IProblemDetailsService for validation errors

marcominerva opened the pull request:
This commit adds integration with `IProblemDetailsService` to return validation errors as a **ProblemDetails** response in the ASP.NET Core application. The `AddProblemDetails` method is utilized to customize the response, including a summary of validation errors and a timestamp.

marcominerva committed:
Integrate IProblemDetailsService for validation errors

This commit adds integration with `IProblemDetailsService` to return validation errors as a **ProblemDetails** response in the ASP.NET Core application. The `AddProblemDetails` method is utilized to customize the response, including a summary of validation errors and a timestamp.

captainsafia commented on api/Program.cs:
```diff
@@ -16,8 +15,23 @@
+            context.ProblemDetails.Detail = $"Error(s) occurred: {validationProblem.Errors.Values.Sum(x => x.Length)}";
+        }
+
+        context.ProblemDetails.Extensions.TryAdd("timestamp", DateTime.UtcNow.ToString("o", CultureInfo.InvariantCulture));
```
Could I trouble you to add some code here that showcases how this can be used to set the property keys on the errors object to comply with the JSON casing policy defined in serializer options?

It won't be 100% foolproof but should hopefully provide a way for folks to address this.

marcominerva committed:
Enhance JSON options and validation error handling

- Customized `ProblemDetails` response for validation errors to align property keys with JSON casing policy.

captainsafia reviewed (approved):
Thank you, @marcominerva! 😄
